File: 0035-search-insert-position/0035-search-insert-position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int j = nums.size() - 1, i = 0, mid;
        while(i <= j){
            mid = (j + i) / 2;
            if(nums[mid] == target) return mid;
            if(nums[mid] > target) j = mid - 1;
            else if(nums[mid] < target)i = mid + 1;
        }
        if(mid == 0 && nums[mid] > target) return 0;
        if(nums[mid] < target) return mid + 1;
        else if(nums[mid] > target && nums[mid - 1] < target) return mid;

        return mid - 1; 
    }
};

File: 0035-search-insert-position/0035-search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        b = 0
        e = len(nums)
        m = (e + b)//2
        while e > b:
            if nums[m] == target:
                return m
            elif nums[m] > target:
                e = m
            else:
                b = m + 1
            m = (e + b)//2
        return m

        